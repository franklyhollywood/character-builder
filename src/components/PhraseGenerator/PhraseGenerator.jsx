import React from 'react';

export default function PhraseGenerator({
	phrase,
	onSetPhrase,
	showPhrase,
	onSetShowPhrase,
}) {
	return (
		<div>
			<br />
			{/* need to add text input field
			    need to add button 
				need to add state to text field */}
			Add da catchphrase:
			<form
				//Because we are in a form it has to be onSUBMIT, and
				// you have to prevent default.
				onSubmit={(e) => {
					e.preventDefault();
					onSetShowPhrase((currentState) => [...currentState, phrase]);
				}}
			>
				<input
					type='text'
					value={phrase}
					onChange={(e) => onSetPhrase(e.target.value)}
				/>
				<button>Add</button>
			</form>
			<ul>
				{showPhrase.map((phrase) => (
					<li key={phrase}>{phrase}</li>
				))}
			</ul>
		</div>
	);
}
