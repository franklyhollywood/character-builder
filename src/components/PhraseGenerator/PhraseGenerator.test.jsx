import { screen, render } from '@testing-library/react';
import PhraseGenerator from './PhraseGenerator';

it('should render the catchphrase input', () => {
	const { container } = render(
		<PhraseGenerator showPhrase={['tree', 'forest']} />
	);
	expect(container).toMatchSnapshot();
});
