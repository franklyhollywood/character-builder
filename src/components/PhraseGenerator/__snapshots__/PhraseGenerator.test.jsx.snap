// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the catchphrase input 1`] = `
<div>
  <div>
    <br />
    Add da catchphrase:
    <form>
      <input
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
    <ul>
      <li>
        tree
      </li>
      <li>
        forest
      </li>
    </ul>
  </div>
</div>
`;
